src/na/na_ofi.c

```diff
@@ -591,6 +604,58 @@ na_ofi_addr_ht_lookup(na_class_t *na_class, struct na_ofi_reqhdr *reqhdr,
     return ret;
 }
 
+/* for psm2 on an RPC origin */
+static na_return_t
+na_ofi_addr_ht_lookup_psm2(na_class_t *na_class, const char *name,
+                           fi_addr_t *src_addr)
+{
+    na_uint64_t addr_key;
+    fi_addr_t new_src_addr;
+    na_return_t ret;
+
+    addr_key = psm2_straddr_2_key(name);
+
+    /* for psm2 name is in the "psm2://fi_addr_psmx2://40302:0" style */
+    ret = na_ofi_addr_ht_lookup(na_class, name + 7, NULL, addr_key, &new_src_addr);
```

Comment:
I don't grok how it works here? for example the first time you pass "fi_addr_psmx2://40302:0" as node, NULL as service to na_ofi_addr_ht_lookup inside that when not hit hash-table, then na_ofi_av_insert that {node, service}, can the fi_av_insertsvc work for this {node, service}?

Reply:
BTW, did you test with "export FI_PSM2_NAME_SERVER=0" (or unset that ENV), can it works? (better to test with different mode that create multiple SEPs, or one SEP with multiple contexts).

Reply:
from my experiments fi_av_insertsvc() doesn't work, the fi_getinro()/fi_av_insert() pair works. I tested with export FI_PSM2_NAME_SERVER=0, one SEP with multiple endpoints and multiple basic endpoints work. Did not test multiple SEPs.